Adyen/Service/Checkout.cs

```diff
@@ -55,7 +57,7 @@ public Checkout(Client client) : base(client)
             _paymentDetails = new PaymentDetails(this);
             _paymentSession = new PaymentSession(this);
             _paymentsResult = new PaymentsResult(this);
-            _paymentLinksResult = new PaymentLinks(this);
+            _paymentLinksResult = new PaymentLinks(this, null);
```

Comment:
This might got added due to an upmerge with develop / so the review here may be out-of-scope. @AlexandrosMor @jillingk 

Anyway, I noticed that we're passing a `null` parameter here. So I did some digging as why we would pass `null` here.

Looking at `PaymentLinks.cs`, the parameter is called `endpoint`, which should be named to `linkId` (according to the unittests). Let's rename this for consistency first. Thinking of a use-case, this `linkId` should probably be variable and I recommend not instantiating this when CheckOut is created (unless you want to instantiate a new one every request?)

I think it would be better to pass this linkId elsewhere by restructuring this. This seems like a class where you send a specific request, and then parse its results. Looking at the `getPaymentLinksAsync(...)`, `getPaymentLinks(...)`, `patchPaymentLinks(...)` and `patchPaymentLinksAsync(...)` methods. We will update the request with the right url by appending `/` + `linkId`  

Suggestions:
- [ ] (In a separate ticket*), refactor this `null`-parameter to be outside of its ctor (or don't instantiate it as a memberVariable)
- [ ] Rename the camelCased methods `**G**etPaymentLinksAsync(...)`, `**G**etPaymentLinks(...)`, `**P**atchPaymentLinks(...)` and `**P**atchPaymentLinksAsync(...)` to UpperCased following C# coding guideliness.

Reply:
Good one about the camelcase (in the other libs we always use camelcase hence my confusion here)

Reply:
The reason I pass that null there is because we use that resource class to create the request url. In case of the Patch and the Get, we need to pass path parameters hence I'm adding that there. It's not a nice way to do it like that, but I just wanted to quickly test the Patch/Get request with the client which is why I did it like that. We'll be sure to refactor this later though when we regenerate the Checkout models and update the services!